### PR TITLE
Projects Pages and Inventory

### PR DESCRIPTION
**Pull request by ConnorHibbs**
This pull request addresses the process of finding another user agent in the system, viewing more information about them, and finally viewing their inventory- including details about specific items.

* Fixed errors preventing project pages from rendering,
* Changed project-type bar to buttons which will be used to filter items in a future PR
* Fixed errors preventing user agent page from rendering
* Added details modal for each inventory item
* Detail modal can display more information about items that is not immediately relevant

### CODE REVIEW

**Review by bensonjohn — APPROVED** [review]

File: packages/app/pages/AllProjectsPage/AllProjectsPage.tsx [hunk 1]
```diff
@@ -6,10 +6,12 @@
  * @since:   2017-08-07
  */
 
-import * as React from 'react'
-import * as themeable from 'react-themeable'
-import BindAgent, { AllOrgsType } from '@vflows/bindings/agent/allOrganizations'
-import Link from '@vflows/views/atoms/Link'
+import * as React from "react"
```
Comment (fontainesw) [inline packages/app/pages/AllProjectsPage/AllProjectsPage.tsx:9]: Why did you change the syntax here?

File: packages/app/pages/AllProjectsPage/AllProjectsPage.tsx [hunk 2]
```diff
@@ -19,43 +21,140 @@ interface Props {
...
+   * Handler function for when buttons are pressed.
+   * TODO apply the filter to the user agents
+   * @param buttonName the text of the button that was clicked
+   */
+  handleClick = (buttonName) => {
+    this.setState({ activeButton: buttonName })
+    this.forceUpdate(() => console.log('Finished Updating'))
+    // this.state.activeButton  = buttonName
+    console.log("Clicked", buttonName)
+    // TODO refresh the list and filter
```
Comment (fontainesw) [inline packages/app/pages/AllProjectsPage/AllProjectsPage.tsx:52]: Just a question: What does this mean?

Comment (ConnorHibbs) [reply packages/app/pages/AllProjectsPage/AllProjectsPage.tsx:52]: What does "this" mean? 😛

Comment (fontainesw) [reply packages/app/pages/AllProjectsPage/AllProjectsPage.tsx:52]: The TODO

File: packages/app/pages/AllProjectsPage/AllProjectsPage.tsx (hunk 1 above)
Comment (ConnorHibbs) [reply packages/app/pages/AllProjectsPage/AllProjectsPage.tsx:9]: I didn't. It just changed to double quotes to align with the linter

**Review by fontainesw — APPROVED** [review]
Looks good

File: packages/app/pages/AllProjectsPage/AllProjectsPage.tsx (hunk 2 above)
Comment (ConnorHibbs) [reply packages/app/pages/AllProjectsPage/AllProjectsPage.tsx:52]: The current projects page has buttons on it to filter by different types of user agents ("All", "Cooperative", "Projects", "Organizations", "Groups"). The buttons were not functional (they were just text) at the start of the spike. As part of the spike they were turned into buttons and an action listener was attached, but the filtering functionality wasn't implemented yet. It has not been in scope of a sprint since, so the TODO just marks that the action still needs to happen but it hasn't been scheduled.
